pkg/client/client_test.go

```diff
@@ -107,9 +111,94 @@ func (TestConfig) GetClientVersion() string {
 	return "1.0.0"
 }
 
+func TestAddandRemoveOnTrack(t *testing.T) {
```

Comment:
Can you create a test suite for these new tests to have better code reuse? 

We already have some test suites towards the end like this:

func TestClientTestSuite(t *testing.T) {
	suite.Run(t, new(ClientTestSuiteAB))
	suite.Run(t, new(ClientTestSuiteFM))
}

So you can add 1 test suite for the notifications APIs and another one for event tracking for the tests you added further below. Don't worry about refactoring the current track tests, you can simply add the track notification track tests in the test suite